iwyu_path_util.cc

```diff
@@ -190,18 +214,26 @@ string ConvertToQuotedInclude(const string& filepath) {
   for (Each<HeaderSearchPath> it(&search_paths); !it.AtEnd(); ++it) {
     // All header search paths have a trailing "/", so we'll get a perfect
     // quoted include by just stripping the prefix.
-    if (StripLeft(&path, it->path)) {
-      if (it->path_type == HeaderSearchPath::kSystemPath)
-        return "<" + path + ">";
-      else
-        return "\"" + path + "\"";
-    }
+
+    if (!StripPathPrefix(&path, it->path))
+      continue;
+    else if (it->path_type == HeaderSearchPath::kSystemPath)
+      return "<" + path + ">";
+    else
+      return "\"" + path + "\"";
   }
 
-  // Case 2: Uses the implicit "-I." entry on the search path.  Always local.
+  // Case 2: Uses the implicit "-I <basename current file>" entry on the search path
+  if (!includer_path.empty())
+      StripLeft(&path, includer_path);
```

Comment:
It looks like this will only work if `includer_path` is absolute -- can you add a:

    CHECK_(includer_path.empty() || IsAbsolutePath(includer_path));

to the top of this method?